File: sdks/csharp/src/NovaBilling/Analytics/Requests/GetMrrBreakdownAnalyticsRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record GetMrrBreakdownAnalyticsRequest
{
    [JsonIgnore]
    public string? DateFrom { get; set; }

    [JsonIgnore]
    public string? DateTo { get; set; }

    [JsonIgnore]
    public string? Currency { get; set; }

    [JsonIgnore]
    public GetMrrBreakdownAnalyticsRequestGroupBy? GroupBy { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/analytics/types/get_mrr_breakdown_analytics_request.rb
# frozen_string_literal: true

module Novabilling
  module Analytics
    module Types
      class GetMrrBreakdownAnalyticsRequest < Internal::Types::Model
        field :date_from, -> { String }, optional: true, nullable: false, api_name: "dateFrom"
        field :date_to, -> { String }, optional: true, nullable: false, api_name: "dateTo"
        field :currency, -> { String }, optional: true, nullable: false
        field :group_by, -> { Novabilling::Analytics::Types::GetMrrBreakdownAnalyticsRequestGroupBy }, optional: true, nullable: false, api_name: "groupBy"
      end
    end
  end
end
